doc/user-manual/getting-started/hello-world.lagda.rst

```diff
@@ -7,35 +7,417 @@
+for the standard ``Vec``, but any other valid identifier will work, like
+``Vector`` or ``V``.
+
+``Vec A n`` is the family of types that represent the collection
+of vector spaces. This family of types is indexed by the dimension, i.e. by
+objects of ``Nat`` (the set of natural numbers). The components of the vector
+can belong to an arbitrary element type that we represented with the identifier
+``A`` (again, ``A`` could have been ``X`` or any other valid identifier; ``n``
+could have been ``m`` and so on).
+
+We refer to ``A`` as the *parameter* of the datatype ``Vec A n``. So far, we
+know that the index ``n`` is a natural number, but we also need to know what
+is ``A``. We will specify that A can be any element type of the sort ``Set``,
+but it could have been any other sort with a greater level of abstraction. We
+express the fact that ``A`` has type ``Set`` or ``A`` ranges over ``Set`` by
+means of the colon ``:``.
+
+We can infer that ``Vec A n`` will also range over ``Set`` (the lower upper
+bound of the types involved). We are now entitled to write the signature of
+our datatype: ``Vec (A : Set) : ℕ → Set``.
+
+To declare the datytype, we place its signature between the keywords ``data``
+and ``where``.
+
+The constructors ``[]`` and ``_∷_``
+-----------------------------------
+
+Constructors are declared in new lines and indented with two spaces.
+
+We chose for the first constructor the identifier ``[]``. It represents the
+empty vector, and its type is the vector space of length ``0``.
+
+The second constructor is a :ref:`mixfix operator <mixfix-operators>` named
+``_∷_`` (pronounced append). For all numbers, it takes as input an element
+of ``A`` and a vector. As output, it produces a vector with a length
+increased by one.
+
+The declaration with keyword ``ìnfixr`` does not belong to the datatype
+declaration; therefore it is not indented. It establishes the
+:ref:`precedence <precedence>` of the operator append. This finishes our
+explanation of the ‘Hello world’ dependent type example.
+
+‘Hello world’ proof
+===================
+
+We will see in this section how to enunciate and prove with Agda the
+associativity of natural numbers under the addition. This time we will build
+the code line by line. You can load the file in each step.
+
+Enunciation of associativity
+----------------------------
+
+Please create a new ``.agda`` file named ``hello-world-proof.agda``.
+Paste or type the following code:
+
+.. code-block:: agda
+
+  module hello-world-proof where
+
+Loading the file with ``C-c C-l`` should work. This is fon now the empty
+module. Now we bring to scope the datatype ``Nat`` and the addition operation
+``_+_``, both defined in the Agda Builtin library.
+
+.. code-block:: agda
+
+  open import Agda.Builtin.Nat using (Nat ; _+_)
+
+Next, we import the propositional equality between two terms ``_≡_``. This will
+allow us to state that two terms compute to the same normal form.
```

Comment:
You are confusion propositional and definitional equality here. Definitional equality states (to a first approximation) that the terms compute to the same normal form, but cannot be presented internally in Agda. Propositional equality (which you are using here) represents that Agda can *prove* that the two terms are equal. For example, `m + n` and `n + m` are not definitionally equal but they are propositionally equal.

Reply:
Yes, thanks. I added this description. But if the result of evaluating the propositional equality between two terms is `refl`, it means that they are definitionally equal, is it so?